src/NCalc/Expression.cs

```diff
@@ -291,49 +288,44 @@ public object Evaluate()
             if ((Options & EvaluateOptions.IterateParameters) == EvaluateOptions.IterateParameters)
             {
                 int size = -1;
-                ParametersBackup = new Dictionary<string, object>();
-                foreach (string key in Parameters.Keys)
-                {
-                    ParametersBackup.Add(key, Parameters[key]);
-                }
-
-                ParameterEnumerators = new Dictionary<string, IEnumerator>();
 
                 foreach (object parameter in Parameters.Values)
                 {
                     if (parameter is IEnumerable enumerable)
                     {
-                        int localsize = 0;
+                        int localSize = 0;

```

Comment:
This would be cleaner if it used the linq extension method.

`int localSize = enumerable.Count();`

Reply:
Unfortunately `IEnumerable` class has no such method. We can replace it with `ICollection` type which has `Count` property

Reply:
ICollection would be a breaking change as there's no guarantee someone hasn't used an IEnumerable only as a parameter.  Ignore this suggestion, I didn't realize the extension was only on `IEnumerable<T>`